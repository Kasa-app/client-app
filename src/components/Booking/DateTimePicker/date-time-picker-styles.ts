import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  inputContainer: {
    borderWidth: 1,
    width: '100%',
    marginTop: 30,
    paddingStart: '5%',
    minHeight: 50,
    justifyContent: 'center',
    borderRadius: 12,
  },
});

export default styles;
