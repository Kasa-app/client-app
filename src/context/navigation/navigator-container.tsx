import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import LoginPage from '../../screens/authentification/login/login-page-view';
import SignUpPage from '../../screens/authentification/signup/signup-page';
import StartupPage from '../../screens/authentification/startup/startup-page';
import ServiceDetailsPage from '../../screens/service-details/service-details-page';
import { TabsNavigation } from '../../components/Bottom-Navigation-View/Tabs-navigation/Bottom/tabs-navigation-view';
import ServiceList from '../../screens/service-list/service-list-page';
import SearchJobbersPage from '../../screens/search-jobbers/search-jobbers';
import JobberProfilePage from '../../screens/jobber-profile/jobber-profile';
import BookingPage from '../../screens/booking/booking-page';
import LaunchPage from '../../screens/launch-page/launch-page';
import Chat from '../../components/Bottom-Navigation-View/Chat/Chat-View/chat';

const Stack = createNativeStackNavigator();

export const NavigatorContainer = () => {
    return (
        <NavigationContainer>
            <Stack.Navigator
                screenOptions={{
                    headerShown: false,
                    presentation: 'modal',
                    animationTypeForReplace: 'push',
                    animation: 'slide_from_bottom',
                    animationDuration: 200
                }}
                initialRouteName='LaunchScreen'
            >
                <Stack.Screen name="LaunchScreen" component={LaunchPage} />
                <Stack.Screen name="StartupScreen" component={StartupPage} />
                <Stack.Screen name="LoginScreen" component={LoginPage} />
                <Stack.Screen name="SignUpScreen" component={SignUpPage} />
                <Stack.Screen name="ServiceDetailsScreen" component={ServiceDetailsPage} />
                <Stack.Screen name="BookingScreen" component={BookingPage} />
                <Stack.Screen name="ChatScreen" component={Chat} />
                <Stack.Screen name="TabsNavigationScreen" component={TabsNavigation} />
                <Stack.Screen name="ServiceListScreen" component={ServiceList} />
                <Stack.Screen name="SearchJobbersScreen" component={SearchJobbersPage} />
                <Stack.Screen name="JobberProfileScreen" component={JobberProfilePage} />
            </Stack.Navigator>
        </NavigationContainer>
    );
};
