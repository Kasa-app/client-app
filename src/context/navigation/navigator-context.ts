import { createContext } from "react";
import { NavigatorContextType } from "../../models/contextType/NavigatorContextType";

export const NavigatorContext = createContext<NavigatorContextType>({
  activeStep: null,
  setActiveStep: () => { },
});