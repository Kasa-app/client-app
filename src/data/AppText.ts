export const APP_TEXT = {
    authentification: {
        startup: {
            welcome_text: "Bienvenue sur Kasa",
            pickupLine1: "Gagnez du temps, √©conomisez de l'√©nergie",
            pickupLine2: "Commencez d√®s aujourd'hui !",
            inscription_text: "S'inscrire",
            connexion_text: "Se connecter",
        },
        signUp: {
            create_account_text: 'Cr√©ez un compte',
            ask_email_text: 'Quel est votre courriel ?',
            ask_user_name_text: 'Quel est votre nom et pr√©nom ?',
            ask_user_birthdate: 'Quel est votre date de naissance ?',
            ask_user_password: 'Quel mot de passe utiliserez-vous sur Kasa ?',
            email_signUp_text: 'S\'inscrire avec un email',
            social_media_signUp_text: 'ou inscrivez-vous avec',
            already_have_account_text: 'Appuyez ici si vous avez d√©ja un compte !',
            continue_text: 'Continuer',
            create_account_button_text:'Cr√©er mon compte',
            checkBox_text: 'J\'ai lu et j\'accepte les conditions g√©n√©rales d\'utilisation \n ainsi que la politique de protection des donn√©es personnelles',
            email_place_holder_text: 'johndoe@gmail.com',
            lastName_place_holder_text: 'Nom',
            firstName_place_holder_text: 'Pr√©nom',
            birthdate_place_holder_text: '2023-01-01',
            birthdate_restriction_text: 'Vous devez avoir au moins 18 ans pour acc√©der √† Kasa',
            password_place_holder_text: 'Mot de passe',
            password_not_match_error:'Les mots de passes ne correspondent pas',
            password_spec_error: 'Le mot de passe ne respecte pas les crit√®res ci-dessous',
            email_error: 'Adresse courriel invalide',
            email_already_exist:'Votre adresse courriel existe d√©ja',
            validate_email_error: 'Une erreur est survenue lors de la validation de votre adresse courriel',
            name_input_error:'Le champ doit avoir entre 3 et 20 caract√®res et ne doit contenir aucun caract√®re sp√©cial',
            confirmPassword_place_holder_text: 'Confirmer le mot de passe',
            password_restriction_text:
                `Le mot de passe doit contenir au moins:
                
- Une lettre majuscule. ex: B
- Une lettre minuscule. ex: c
- Un chiffre. ex: 1
- Le mot de passe doit contenir au moins 8 caract√®res.
            
exemple de mot de passe: Azerty123456
            `
      ,

    },
    signIn: {
      question_forget_password: 'Mot de passe oubli√© ?',
      social_media_signIn_text: 'ou connectez-vous avec',
      not_have_account_text: "Appuyez ici si vous n'avez pas de compte !",
      message_bienvenue: "Bienvenue",
      adresse_email: "Adresse courriel",
      mot_de_passe: "Mot de passe",
      connexion: "Connexion",
      login_failed: "La connexion a √©chou√©. Veuillez v√©rifier vos informations de connexion et r√©essayer.",
    },
  },
  booking: {
    address: 'Adresse',
    back_to_home: 'Retourner √† l\'accueil',
    booking_error: 'Une erreur est survenue lors de la r√©servation',
    postal_code: 'Code postal',
    continue: 'Continuer',
    comments: 'Commentaires',
    no_comment: "Il n'y a aucun commentaire pour le moment",
    search_jobber: 'Rechercher des jobbers',
    services_done: 'A r√©alis√© plus de',
    phone_number_placeholder: '819-XXX-XXXX',
    services: 'services',
    see_profile: 'Voir le profil',
    member_since: 'Membre depuis',
    no_jobbers: 'Malheureusement, il n\'y a aucun jobber disponible pour le moment. Veuillez r√©essayer plus tard.',
    languages: 'Langues parl√©es:',
    to_book: 'R√©servez',
    skills: 'Habilet√©s et exp√©riences',
    booking_success: 'Votre r√©servation a √©t√© effectu√©e avec succ√®s!',
    steps: {
        photos: 'Souhaitez-vous ajouter des photos ? (optionnel)',
        estimated_duration: 'Estimation de la dur√©e de service',
        address: 'Adresse de la prestation',
        pickupAddress: "Quel est l'adresse de d√©part ?",
        date: 'Quel jour vous convient le mieux ?',
        drop_address: "Quel est l'adresse d'arriv√©e ?",
        come_with_truck: "Le jobber doit-il venir avec son camion ?",
        own_vehicle: "Le jobber doit-il poss√©der un v√©hicule ?",
        own_mower: "Le jobber doit-il apporter sa propre tondeuse ?",
        frequency: "Fr√©quence de la demande ?",
        keep_duration: "Quand commence et se termine la garde ?",
        phone_number: "√Ä quel num√©ro de t√©l√©phone le jobber pourra-t-il vous joindre ?",
        more_details: "D√©tails suppl√©mentaires (optionnel)",
        jobber_counts: "Combien de prestataires vous faut-il ?",
        time: "Quelle heure vous convient le mieux ?",
        warning_time_min:"L'heure de d√©but doit √™tre apr√®s ou √† 6h du matin",
        warning_time_max:"L'heure de fin doit √™tre avant ou √† 21h00",
        package_delivery: "√Ä quel adresse le colis doit-il √™tre livr√© ?",
        more_than_one_day: "Envisagez-vous une dur√©e de service sup√©rieure √† une journ√©e ?",
        date_debut: "√Ä quel date souhaitez-vous que le service d√©bute ?",
        date_fin: "√Ä quel date souhaitez-vous que le service prenne fin ?"
    },

    stepKey:{
      jobber_counts: "jobberCounts",
      estimated_duration: "estimatedDuration",
      more_than_one_day:"moreThanOneDay"
    }
	},
  components: {
    show_more: 'Voir plus',
    show_less: 'Voir moins',
    retry: 'R√©essayer',
  },
  errors: {
    service_details: 'Une erreur innatendue est survenue lors du chargement des d√©tails du service.',
  },
  profile: {
    profile: 'Profil',
    manage_account: 'G√©rer mon compte',
    notifications: 'Notifications',
    options: 'Options',
    member_since: 'Membre depuis',
    help: 'Aide et assistance',
    logout: 'Se d√©connecter'
  },
  maintenance: {
    title: 'Sous d√©veloppement',
    message: "Oops! üõ†Ô∏è Ce service est en cours de d√©veloppement et n'est pas disponible pour le moment. Nous travaillons fort pour le rendre disponible sous peu. Revenez bient√¥t !",
  },
  services: {
    estimation_taux_horaire: 'Estimation du taux horaire:',
    estimation_duree_service: 'Estimation de la dur√©e du service:',
    said: 'a dit:',
    start_booking: 'Commencer ma r√©servation',
  },
  homePage: {
    categorie_section: 'Nos univers de categories',
    popular_services_section: 'Nos services populaires',
    recommended_services_section: 'Services recommand√©s',
    search_placeHolder: 'Rechercher un service'
  },
  history:{
    reservations_title:'Mes r√©servations',
    service_name: 'Nom du service',
    date_hour_title:'Date et heure',
    adresse_title:'Adresse',
    button_title:'G√©rer ma demande'
  }
}