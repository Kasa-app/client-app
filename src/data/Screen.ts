export const Screen = {
  authentification: {
    login: 'LoginScreen',
    startup: 'StartupScreen',
    signUp: 'SignUpScreen',
  },
  booking: 'BookingScreen',
  searchJobbers: 'SearchJobbersScreen',
  jobberProfile: 'JobberProfileScreen',
  serviceDetails: 'ServiceDetailsScreen',
  tabsNavigation:'TabsNavigationScreen',
  serviceList:'ServiceListScreen',
  chat:'ChatScreen'
};