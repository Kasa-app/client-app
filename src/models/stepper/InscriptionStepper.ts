export enum InscriptionStepper {
  SignUpWithEmail = 'signUpWithEmail',
  UserEmail = 'User Email',
  UserName = 'User Name',
  UserBirthdate = 'User birthdate',
  UserPassword = 'User password',
}