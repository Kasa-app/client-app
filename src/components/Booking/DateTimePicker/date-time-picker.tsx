import React, { useCallback, useState } from "react";
import { View, TouchableOpacity, Text } from "react-native";
import DateTimePicker, { DateTimePickerEvent } from '@react-native-community/datetimepicker';
import styles from './date-time-picker-styles';
import { HandleContinueBtn } from "../../../hooks/BookingContinueBtnHooks";

type Props = {
  onChange: (value: any) => void;
  value: string;
};
const DateTimePickerInput = ({ onChange, value }: Props) => {
  const [showDatePicker, setShowDatePicker] = useState(false);
  const { enableBtn } = HandleContinueBtn();
  const padZero = (num: number) => (num < 10 ? `0${num}` : num);

  const handleChange = useCallback((event: DateTimePickerEvent, selectedDate: Date | undefined) => {
    setShowDatePicker(false);
    const day = padZero(selectedDate!.getDate());
    const month = padZero(selectedDate!.getMonth() + 1);
    const year = selectedDate!.getFullYear();
    const formattedDate = `${year}-${month}-${day}`;
    onChange(formattedDate);
    enableBtn();
  }, [showDatePicker]);

  const getCurrentDate = () => {
    const date = new Date();
    const year = date.getFullYear();
    const month = (date.getMonth() + 1).toString().padStart(2, '0');
    const day = date.getDate().toString().padStart(2, '0');
    return `${year}-${month}-${day}`;
  };

  const showDatepicker = () => {
    setShowDatePicker(true);
  };

  return (
    <View>
      <TouchableOpacity onPress={showDatepicker} style={styles.inputContainer}>
        <Text>{value || getCurrentDate()}</Text>
      </TouchableOpacity>
      {showDatePicker && (
        <DateTimePicker
          mode='date'
          value={new Date()}
          minimumDate={new Date()}
          is24Hour={true}
          onChange={handleChange}
        />
      )}
    </View>
  );
};

export default DateTimePickerInput;
