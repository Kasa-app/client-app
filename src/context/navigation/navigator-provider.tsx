import React, { useState } from 'react';
import { NavigatorContext } from './navigator-context';
import { InscriptionStepper } from '../../models/stepper/InscriptionStepper';
import { NavigatorContextType } from '../../models/contextType/NavigatorContextType';

export const NavigatorProvider = ({ children }: any) => {
  const [activeStep, setActiveStep] = useState(InscriptionStepper.SignUpWithEmail);
  const context: NavigatorContextType =
  {
    activeStep,
    setActiveStep,
  };

  return (
    <NavigatorContext.Provider value={context}>
      {children}
    </NavigatorContext.Provider>
  );
};

