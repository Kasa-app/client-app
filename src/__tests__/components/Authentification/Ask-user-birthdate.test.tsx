import { Text } from 'react-native';
import React from 'react';
import { it } from '@jest/globals';
import { create, act } from 'react-test-renderer';
import { render } from '@testing-library/react-native';
import AskUserBirthdate from '../../../components/Authentification/Ask-user-birthdate/Ask-user-birthdate';
import DateTimePicker from '@react-native-community/datetimepicker';
import ButtonWithBackgroundColor from '../../../components/Button-with-color-background/button';
import { InscriptionStepper } from '../../../models/stepper/InscriptionStepper';
import { NavigatorContext } from '../../../context/navigation/navigator-context';
import { Screen } from '../../../data/Screen';
import { fireEvent } from 'react-native-testing-library';
jest.mock('react-native-keyboard-aware-scroll-view', () => {
  return {
    KeyboardAwareScrollView: ({ children }: { children: React.ReactNode }) => <div>{children}</div>,
  };
});

const mockNavigatorContextValue = {
  routeName: '',
  setRouteName: jest.fn(),
  activeStep: InscriptionStepper.UserEmail,
  setActiveStep: jest.fn(),
};
const MockNavigatorContextProvider = ({ children }: any) => (
  <NavigatorContext.Provider value={mockNavigatorContextValue}>
      {children}
  </NavigatorContext.Provider>
);

describe('AskUserBirhtdateComponent', () => {

  let tree: any

  beforeEach(() => {
    tree = render(<AskUserBirthdate />);
  });

  it('snapshot', () => {
    expect(tree).toMatchSnapshot();
  });

  it('When birthdate input is pressed should show calendar', () => {
    const textComponent = tree.root.findAllByType(Text)
    const birthdateText = textComponent[0].props;
    act(() => birthdateText.onPress());

    expect(tree.root.findByType(DateTimePicker)).toBeTruthy();
  });

  it('should set active step with userPassword', () => {
    const component = render(<AskUserBirthdate />, { wrapper: MockNavigatorContextProvider });
    const button = component.root.findByType(ButtonWithBackgroundColor).props;

    act(() => button.handlePress());
    expect(mockNavigatorContextValue.setActiveStep).toHaveBeenCalledWith(InscriptionStepper.UserPassword);
});
});