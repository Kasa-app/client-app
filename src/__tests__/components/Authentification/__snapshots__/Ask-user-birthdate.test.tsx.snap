// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AskUserBirhtdateComponent snapshot 1`] = `
<div>
  <RNCSafeAreaView
    edges={
      {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
          "marginTop": "10%",
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          [
            {
              "borderColor": "#E67918",
              "borderRadius": 12,
              "borderWidth": 1,
              "color": "gray",
              "height": "20%",
              "paddingStart": "5%",
              "paddingTop": "2%",
              "width": "90%",
            },
            {
              "color": "gray",
            },
          ]
        }
      >
        2023-01-01
      </Text>
      <Text
        style={
          {
            "color": "gray",
            "fontFamily": "AvenirNext-Regular",
            "fontSize": 15,
          }
        }
      >
        Vous devez avoir au moins 18 ans pour accéder à Kasa
      </Text>
      <View
        style={
          {
            "marginTop": "15%",
          }
        }
      >
        <View
          style={
            [
              {
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignContent": "center",
                "alignItems": "center",
                "backgroundColor": "#E2B288",
                "borderRadius": 4,
                "display": "flex",
                "height": undefined,
                "justifyContent": "center",
                "minWidth": "35%",
                "opacity": 1,
                "paddingVertical": 7,
                "width": "77%",
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "black",
                    "fontFamily": "AvenirNext-Regular",
                    "fontSize": 13,
                    "textAlign": "center",
                  },
                  {
                    "color": "gray",
                  },
                  undefined,
                ]
              }
            >
              Continuer
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</div>
`;
